src/main/java/org/opensearch/securityanalytics/threatIntel/service/SATIFSourceConfigService.java

```
@@ -197,34 +198,59 @@ private String getIndexMapping() {
      *
      * @param stepListener setup listener
      */
     public void createJobIndexIfNotExists(final StepListener<Void> stepListener) {
         // check if job index exists
         if (clusterService.state().metadata().hasIndex(SecurityAnalyticsPlugin.JOB_INDEX_NAME) == true) {
-            stepListener.onResponse(null);
-            return;
-        }
-        final CreateIndexRequest createIndexRequest = new CreateIndexRequest(SecurityAnalyticsPlugin.JOB_INDEX_NAME).mapping(getIndexMapping())
-                .settings(SecurityAnalyticsPlugin.TIF_JOB_INDEX_SETTING);
-        StashedThreadContext.run(client, () -> client.admin().indices().create(createIndexRequest, ActionListener.wrap(
-                r -> {
-                    log.debug("[{}] index created", SecurityAnalyticsPlugin.JOB_INDEX_NAME);
+            try {
+                // Check if job index contains old mapping, if so update index mapping (current version = 2)
+                if (shouldUpdateIndex(clusterService.state().metadata().index(SecurityAnalyticsPlugin.JOB_INDEX_NAME), getIndexMapping())) {
+                    log.info("Old schema version found for [{}] index, updating the index mapping", SecurityAnalyticsPlugin.JOB_INDEX_NAME);
+                    IndexUtils.updateIndexMapping(
+                            SecurityAnalyticsPlugin.JOB_INDEX_NAME,
+                            getIndexMapping(), clusterService.state(), client.admin().indices(),
+                            ActionListener.wrap(
+                                    r -> {
+                                        log.info("Successfully updated index mapping for [{}] index", SecurityAnalyticsPlugin.JOB_INDEX_NAME);
+                                        stepListener.onResponse(null);
+                                    }, e -> {
+                                        log.error("Failed to update job index mapping", e);
+                                        stepListener.onFailure(e);
```

maybe failure is due to concurrent modification exception because of another data node doing the same check+update. Can we check the index mapping version again one failure scenario and if version is updated we can ignore error and mark listener as succeeded?

---

added a check for concurrent modification exception